src/main/java/io/antmedia/websocket/WebSocketCommunityHandler.java

```diff
@@ -296,23 +298,49 @@ public void sendMessage(String message, final Session session) {
 		}
 	}
 	
-	public void sendRoomInformation(JSONArray jsonStreamArray , String roomId) 
+	public void sendRoomInformation(Map<String,String> streamDetailsMap , String roomId) 
 	{
 		JSONObject jsObject = new JSONObject();
+		JSONArray jsonStreamIdArray = null;
+		JSONArray jsonStreamNameArray = null;
+		
+		if(streamDetailsMap != null) {
+			jsonStreamIdArray = new JSONArray();
+			jsonStreamNameArray = new JSONArray();
+			for (Map.Entry<String, String> e : streamDetailsMap.entrySet()) {
+				jsonStreamIdArray.add(e.getKey());
+				jsonStreamNameArray.add(e.getValue());
+			}
+		}
+        
 		jsObject.put(WebSocketConstants.COMMAND, WebSocketConstants.ROOM_INFORMATION_NOTIFICATION);
-		jsObject.put(WebSocketConstants.STREAMS_IN_ROOM, jsonStreamArray);	
+		jsObject.put(WebSocketConstants.STREAMS_IN_ROOM, jsonStreamIdArray);	
+		jsObject.put(WebSocketConstants.STREAM_NAMES_IN_ROOM, jsonStreamNameArray);	
 		jsObject.put(WebSocketConstants.ATTR_ROOM_NAME, roomId);
 		jsObject.put(WebSocketConstants.ROOM, roomId);
 		String jsonString = jsObject.toJSONString();
 		sendMessage(jsonString, session);
 	}
 	
-	public void sendJoinedRoomMessage(String room, String newStreamId, JSONArray jsonStreamArray) {
+	public void sendJoinedRoomMessage(String room, String newStreamId, Map<String,String> streamDetailsMap ) {
 		JSONObject jsonResponse = new JSONObject();
+		JSONArray jsonStreamIdArray = null;
+		JSONArray jsonStreamNameArray = null;
+		
+		if(streamDetailsMap != null) {
+			jsonStreamIdArray = new JSONArray();
+			jsonStreamNameArray = new JSONArray();
+			for (Map.Entry<String, String> e : streamDetailsMap.entrySet()) {
+				jsonStreamIdArray.add(e.getKey());
+				jsonStreamNameArray.add(e.getValue());
+			}
+		}
+		
 		jsonResponse.put(WebSocketConstants.COMMAND, WebSocketConstants.NOTIFICATION_COMMAND);
 		jsonResponse.put(WebSocketConstants.DEFINITION, WebSocketConstants.JOINED_THE_ROOM);
 		jsonResponse.put(WebSocketConstants.STREAM_ID, newStreamId);
-		jsonResponse.put(WebSocketConstants.STREAMS_IN_ROOM, jsonStreamArray);	
+		jsonResponse.put(WebSocketConstants.STREAMS_IN_ROOM, jsonStreamIdArray);	
+		jsonResponse.put(WebSocketConstants.STREAM_NAMES_IN_ROOM, jsonStreamNameArray);	
```

Comment:
I don't think it's good idea to have two separate arrays having related information. It creates an overload in parsing.